clients/privacy-center/packages/fides-consent/src/lib/consent-banner.ts

```diff
@@ -0,0 +1,553 @@
+    return {};
+  }
+
+  try {
+    const body = await response.json();
+    debugLog("Got location response from geolocation API, returning:", body);
+    return body;
+  } catch (e) {
+    debugLog("Error parsing response body from geolocation API, returning {}. Response:", response);
+    return {};
+  }
+}
+
+/**
+ * Determine whether or not the banner should be enabled for the given location
+ */
+const isBannerEnabledForLocation = (location?: UserGeolocation): boolean => {
+  const options = getBannerOptions();
+  if (location === undefined || !location) {
+    debugLog("Location unknown, assume banner must be shown. isBannerEnabledForLocation = true");
+    return true;
+  }
+
+  // Get the user's country
+  if (location.country === undefined || !location.country) {
+    debugLog("Country unknown, assume banner must be shown. isBannerEnabledForLocation = true");
+    return true;
+  }
+
+  if (options.isEnabledCountries) {
+    if (options.isEnabledCountries.includes(location.country)) {
+      debugLog(`Country ${location.country} included in isEnabledCountries, banner must be shown. isBannerEnabledForLocation = true`);
+      return true;
+    }
+    debugLog(`Country ${location.country} not included in isEnabledCountries, banner must be hidden. isBannerEnabledForLocation = false`);
+    return false;
+  }
+
+  debugLog("No location-specific rules matched, assume banner must be shown. isBannerEnabledForLocation = true");
+  return true;
+}
+
+/**
+ * Builds a button DOM element with the given id, class name, and text label
+ */
+const buildButton = (id: string, className: string, label?: string, onClick?: (event: MouseEvent) => void): HTMLButtonElement => {
+  const button = document.createElement("button");
+  button.id = id;
+  button.innerHTML = label || "";
+  button.classList.add("fides-consent-banner-button");
+  button.classList.add(className);
+  button.addEventListener("click", (event) => {
+    debugLog(`Fides consent banner button clicked with id='${id}'`);
+    if (onClick !== undefined) {
+      onClick(event);
+    }
+  });
+  return button;
+};
+
+/**
+ * Show the banner
+ */
+const showBanner = (banner: HTMLDivElement) => {
+  banner.classList.remove("fides-consent-banner-hidden");
+};
+
+/**
+ * Hide the banner (probably because the user selected a consent option)
+ */
+const hideBanner = (banner: HTMLDivElement) => {
+  banner.classList.add("fides-consent-banner-hidden");
+};
+
+/**
+ * Builds the DOM elements for the consent banner (container, buttons, etc.) and
+ * return a single div that can be added to the body.
+ */
+const buildBanner = (defaults: CookieKeyConsent): HTMLDivElement => {
+  const options: ConsentBannerOptions = getBannerOptions();
+
+  // Create the overall banner container
+  const banner = document.createElement("div");
+  banner.id = "fides-consent-banner";
+  banner.classList.add("fides-consent-banner");
+  // TODO: support option to specify top/bottom
```

Comment:
This is simple enough, and worth doing up front to ensure we pick a reasonable options API for how to tweak styles